### PR TITLE
docs(semantic-release): implement an automated changelog

### PR DESCRIPTION
**Pull request by daks**
Implementation of semantic-release on nginx-formula.

WIP for now, until we decide how to number versions. 
I propose 1.0 for current code (with `nginx` and `nginx.ng` states) and 2.0 to the needed change of `nginx.ng` to `nginx` and removal of legacy code.

### CODE REVIEW

**Comment by daks** [comment]
Tests are not enabled, or not active when PR is WIP. 
Passed on my repo https://travis-ci.com/daks/nginx-formula/builds/109511689

**Comment by myii** [comment]
@daks Would it be too much to ask to update the matrix to use the new pre-salted images?  Otherwise the CI run takes over 20 minutes.

**Comment by myii** [comment]
In the comparative spreadsheet across all formulas, we calculated this formula to be `v0.55.0`.  What I like about starting with these numbers is that it gives some info about how much development went in to the formula before `semantic-release`.  `v1.0.0` doesn't suggest very much.  But I'm not forcing this issue, just suggesting.

File: FORMULA [hunk 1]
```diff
@@ -1,8 +1,9 @@
 name: nginx
-os: RedHat, Debian, Ubuntu, Suse, Gentoo
-os_family: RedHat, Debian, Suse, Gentoo
-version: 201507
+os: Debian, Ubuntu, Raspbian, RedHat, Fedora, CentOS, Suse, openSUSE, Gentoo, Funtoo, Arch, Manjaro, Alpine, FreeBSD, OpenBSD, Solaris, SmartOS, Windows, MacOS
+os_family: Debian, RedHat, Suse, Gentoo, Arch, Alpine, FreeBSD, OpenBSD, Solaris, Windows, MacOS
```
Comment (myii) [inline FORMULA]: Both `os` and `os_family` in the `template-formula` were given as a full list that can be selected from.  So best to only capture the platforms which definitely work with this formula.

**Comment by myii** [comment]
Looks good otherwise.  Have you tested that the releases are created in your own fork?

**Comment by daks** [comment]
- Images matrix updated
- version: do you mean we create a 0.5.5 manually before merging this PR?
- about FORMULA os list, I have no idea which are compatible, the only thing I can do is keep only tested versions but it's not complete

**Comment by myii** [comment]
> * Images matrix updated

Excellent work, hopefully it will be much faster now.

> * version: do you mean we create a 0.5.5 manually before merging this PR?

Once your ready, we'll create a `v0.5.5` release to cover the history before `semantic-release`.  Then we'll merge this PR to get it all running.  Have you already tested this in your own fork?

> * about FORMULA os list, I have no idea which are compatible, the only thing I can do is keep only tested versions but it's not complete

The best idea is to stick to what we're _definite_ about.  We don't want to mislead anybody (or SPM, if we get that working).  That may mean leaving the lines how they were before.

**Comment by daks** [comment]
I'll test semantic-release on my fork and keep you informed

**Comment by aboe76** [comment]
@daks, nice work, I agree with @myii about the versioning, v0.5.5 is about the right way to introduce
and then merge nginx with nginx:ng as a v1.0.0 that way everybody knows that a stable nginx-formula is created.

**Comment by daks** [comment]
semantic-release tested with success https://github.com/daks/nginx-formula/releases.

**Comment by myii** [comment]
@daks Excellent work again, thanks.

**Comment by myii** [comment]
@daks Just noticed that the `WIP` commit remained!  That will fail at the `commitlint` stage... but should be able to bypass that in Travis -- not usually recommended!

**Pull request merged**